### PR TITLE
Make build pod nodeSelector, Tolerations, and Affinity configurable

### PR DESCRIPTION
**Pull request by tylerphelan**
- Build pods will no longer determine tolerations

fix https://github.com/pivotal/kpack/issues/621

### CODE REVIEW

File: pkg/apis/build/v1alpha2/build_pod.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ const (
 	BuildLabel                   = "kpack.io/build"
 	DOCKERSecretAnnotationPrefix = "kpack.io/docker"
 	GITSecretAnnotationPrefix    = "kpack.io/git"
+	k8sOSLabel                   = "kubernetes.io/os"
```
Comment (matthewmcnew) [inline pkg/apis/build/v1alpha2/build_pod.go:24]: Nice!

File: pkg/apis/build/v1alpha2/build_pod.go [hunk 2]
```diff
@@ -586,9 +586,9 @@ func (b *Build) rebasePod(secrets []corev1.Secret, images BuildPodImages, config
 		},
 		Spec: corev1.PodSpec{
 			ServiceAccountName: b.Spec.ServiceAccount,
-			NodeSelector: map[string]string{
-				"kubernetes.io/os": "linux",
-			},
+			NodeSelector:       b.nodeSelector("linux"),
+			Tolerations:        b.Spec.Tolerations,
```
Comment (matthewmcnew) [inline pkg/apis/build/v1alpha2/build_pod.go:590]: Should we remove the taints that get passed into this?

Comment (tylerphelan) [reply pkg/apis/build/v1alpha2/build_pod.go:590]: thank you!

File: pkg/apis/build/v1alpha2/build_validation.go [hunk 3]
```diff
@@ -52,6 +53,18 @@ func (bs *BuildSpec) validateImmutableFields(ctx context.Context) *apis.FieldErr
 	return nil
 }
 
+func (bs *BuildSpec) validateNodeSelector(_ context.Context) *apis.FieldError {
+	if len(bs.NodeSelector) == 0 {
+		return nil
+	}
+
+	if _, ok := bs.NodeSelector[k8sOSLabel]; ok {
+		return apis.ErrInvalidKeyName(k8sOSLabel, "nodeSelector", "os is determined automatically")
```
Comment (dumez-k) [inline pkg/apis/build/v1alpha2/build_validation.go:62]: nice!

File: pkg/apis/build/v1alpha2/image_types.go [hunk 4]
```diff
@@ -49,11 +49,24 @@ type ImageSpec struct {
...
-	Build                    *corev1alpha1.ImageBuild          `json:"build,omitempty"`
+	Build                    *ImageBuild                       `json:"build,omitempty"`
 	Notary                   *corev1alpha1.NotaryConfig        `json:"notary,omitempty"`
 	DefaultProcess           string                            `json:"defaultProcess,omitempty"`
 }
 
+// +k8s:openapi-gen=true
+type ImageBuild struct {
+	// +listType
+	Bindings corev1alpha1.Bindings `json:"bindings,omitempty"`
```
Comment (dumez-k) [inline pkg/apis/build/v1alpha2/image_types.go:60]: weird how this is the go fmt preferred format ¯\_(ツ)_/¯

**Review by dumez-k — APPROVED** [review]

File: pkg/apis/build/v1alpha2/image_types.go (hunk 4 above)
Comment (dumez-k) [reply pkg/apis/build/v1alpha2/image_types.go:60]: referring to the alignment between listTypes

Comment (tylerphelan) [reply pkg/apis/build/v1alpha2/image_types.go:60]: yea listType throws it off

**Review by matthewmcnew — CHANGES_REQUESTED** [review]
It looks like the open api needs to be regenerated with runtimeClassName, schedulerName

**Comment by tylerphelan** [comment]
> It looks like the open api needs to be regenerated with runtimeClassName, schedulerName

how do I keep forgetting!

**Review by tomkennedy513 — APPROVED** [review]

**Comment by matthewmcnew** [comment]
Let's wait until the corresponding RFC is merged

**Review by matthewmcnew — APPROVED** [review]

**Pull request merged**